src/jit/flowgraph.cpp

```diff
@@ -17495,6 +17499,12 @@ BasicBlock*         Compiler::fgAddCodeRef(BasicBlock*      srcBlk,
                             break;
 #endif // COR_JIT_EE_VERSION
 
+    // SCK_NONE can appear on range checks that optimization proved will succeed
+    // but which it failed to remove for mechanical reasons; use failfast for
+    // these cases.
+    case SCK_NONE:          helper = CORINFO_HELP_FAIL_FAST;
```

Comment:
OK I see now,  This is for the basic block that gets added.
This block would have previously unconditionally have thrown Range Check
With your change this block will call Fail Fast.

SInce the range checks were removed control flow won't reach this block.

And if it does reach this block calling Fail Fast isn't going to be  worse than any alternative.